CONTRIBUTING.md

```diff
@@ -0,0 +1,97 @@
+# Contributing
+
+## Prerequisites
+
+- It's recommended you install `rustup` to acquire the necessary tooling like `rustc` and `cargo`.
+- Rust version [1.76](https://releases.rs/docs/1.76.0/) is required to build libraries from the SDK.
+- If you use VSCode, we recommend you install these extensions:
+  - [rust-analyzer](https://marketplace.visualstudio.com/items?itemName=rust-lang.rust-analyzer)
+  - [Event Better TOML](https://marketplace.visualstudio.com/items?itemName=tamasfe.even-better-toml)
+  - [CodeLLD](https://marketplace.visualstudio.com/items?itemName=vadimcn.vscode-lldb)
+
+## Generated code
+
+If you want to contribute to a file that is generated (header contains `Code generated by Microsoft (R) AutoRest Code Generator.`), the best approach to open a PR on the initial Swagger/TypeSpec specification, as we can NOT merge a PR on generated code (it would be replaced by next generation). Please visit the [Azure/azure-rest-api-specs repo](https://github.com/Azure/azure-rest-api-specs/) to view and make changes to Azure service API specs.
```

Comment:
```suggestion
If you want to contribute to a file that is generated (header contains `Code generated by Microsoft (R) AutoRest Code Generator.`), the best approach is to open a PR on the TypeSpec specification since we cannot replace generated code that will be replaced when regenerated. Please visit the [Azure/azure-rest-api-specs repo](https://github.com/Azure/azure-rest-api-specs/) to view and make changes to Azure service API specifications.
```

We don't and will not support swagger, nor should public-facing docs use atypical abbreviations when considering other languages e.g., "specs".